lfpse.R

```diff
@@ -87,30 +87,132 @@ if (!is.na(text_terms)) {
+  separate_rows(value, sep = " {~@~} ") |>
+  # arrange so that multi-responses appear alphabetised later
+  arrange(value) |>
+  # bring through the value labels
+  left_join(ResponseReference, by = c(
+    "name" = "QuestionId",
+    "value" = "ResponseCode",
+    "TaxonomyVersion" = "TaxonomyVersion"
+  )) |>
+  # remove the unnecessary columns
+  select(!c(value, Property, LastUpdated, IsActive)) |>
+  # pivot back into columns
+  pivot_wider(
+    id_cols = !any_of(ResponseReference$QuestionId),
+    names_from = name,
+    values_from = ResponseText,
+    # collapse multi-responses into single row per entity
+    values_fn = list(ResponseText = ~ str_c(., collapse = "; "))
+  ) |>
+  group_by(Reference) |>
+  mutate(npatient = max(EntityId)) |>
+  ungroup()
+
+# Testing neonate logic
+if (is_neopaed == "neonate") {
+  print("- Running neonate strategy...")
+  lfpse_neopaed <- lfpse_pre_release |>
+    filter(
+      # First condition: Age between 0 and 28 days OR Age Range between 0-28 days
+      (
+        (P004_days > 0 & P004_days <= 28) |
+          (P007 %in% c("0-14 days", "15-28 days"))
+      ) |
+        (P004_days == 0 &
+           (grepl("(?i)\\bneonat|\\bbaby", paste(L006, L006_Other, sep = "")) |
+              (
+                grepl("(?i)\\bn(?:|\\W)i(?:|\\W)c(?:|\\W)u\\b|\\bn(?:|\\W)n(?:|\\W)u\\b|\\bs(?:|\\W)c(?:|\\W)b(?:|\\W)u\\b|\\bneonat|\\bbaby", paste(`F001 - Describe what happened`, `AC001 - Immediate Actions`, `OT003 - What was the clinical outcome for the patient?`, `A008 - Device Type (Other)`, `A008 - Device Type`, `L006 - Specialty`, `L006_Other - Specialty (Other)`, sep = "")) &
+                  !grepl("(?i)\\badult|\\bold|\\belderly|\\bgeriat", paste(L006, L006_Other, sep = ""))
+              )
+           )
+        )
+      # Second condition: Age is missing and text conditions apply
+      | 
+        # Second condition: Age is missing and grepl conditions apply
+        (
+          is.na(P004_days) & is.na(P007) & 
+            (
+              grepl("(?i)\\bneonat|\\bbaby", paste(L006, L006_Other, sep = "")) |
+                (
+                  grepl("(?i)\\bn(?:|\\W)i(?:|\\W)c(?:|\\W)u\\b|\\bn(?:|\\W)n(?:|\\W)u\\b|\\bs(?:|\\W)c(?:|\\W)b(?:|\\W)u\\b|\\bneonat|\\bbaby", paste(`F001 - Describe what happened`, `AC001 - Immediate Actions`, `OT003 - What was the clinical outcome for the patient?`, `A008 - Device Type (Other)`, `A008 - Device Type`, `L006 - Specialty`, `L006_Other - Specialty (Other)`, sep = "")) &
+                    !grepl("(?i)\\badult|\\bold|\\belderly|\\bgeriat", paste(L006, L006_Other, sep = ""))
+                )
+            )
+        )
+    )
+} else if (is_neopaed == "paed") {
+  print("- Running paediatric strategy")
+  lfpse_neopaed <- lfpse_pre_release |>
+    filter(
+      # First condition: Age between 0 and 3 months and PD04 must be pediatric if age is 0
+      (
+        (P004_days > 0 & P004_days < 6575) | (P007 > "D0" & P007 < "Y18")

```

Comment:
Shouldn't this be 3 months - 18 y? Also be careful with conversion factors - LFPSE uses 31 days for months and 372 days for years (but we know some aren't doing this)